### PR TITLE
Add .NET Core templates

### PR DESCRIPTION
**Pull request by tmds**
@bparees PTAL

Will this sync automatically with openshift/origin?

### CODE REVIEW

**Comment by bparees** [comment]
> Will this sync automatically with openshift/origin?

no, there's no automatica syncing from library to origin.  We manage that semi-manually.

But in this case, we don't ship/carry any .net templates in origin today.  (We do carry the imagestream).  Given that these are community templates, I wouldn't expect us to, i'd expect people who want them to get them from here.

File: community.yaml [hunk 1]
```diff
@@ -191,6 +191,13 @@ data:
         regex: redis
         suffix: centos7
   dotnet:
+    templates:
+      - location: https://raw.githubusercontent.com/redhat-developer/s2i-dotnetcore/master/templates/dotnet-example.json
```
Comment (bparees) [inline community.yaml]: we don't put content in both places, so are these supported(official) or community?

**Comment by bparees** [comment]
@tmds looks like you didn't add the imported files to your commit

File: community.yaml (hunk 1 above)
Comment (tmds) [reply community.yaml]: supported, we update these templates and verify they work for each release

Comment (bparees) [reply community.yaml]: ok, then i'd only add them to the official.yaml

**Comment by tmds** [comment]
@bparees removed community, added imported files

**Comment by tmds** [comment]
@bparees it seems nodejs has same templates both in official and community?

https://github.com/openshift/library/blob/a51af1ccfa5d5b6feeb45e191852acd6b8e7603e/official.yaml#L43-L52

https://github.com/openshift/library/blob/a51af1ccfa5d5b6feeb45e191852acd6b8e7603e/community.yaml#L73-L80

**Comment by bparees** [comment]
> @bparees it seems nodejs has same templates both in official and community?

thanks, looks like we have a few to clean up.

**Comment by tmds** [comment]
ok, will these templates find their way to openshift/origin as part of the semi-manual syncing?

**Comment by tmds** [comment]
There are a number of quickstarts here: https://github.com/openshift/origin/tree/master/examples/quickstarts, which made me think .NET Core templates should be there too.

**Comment by tmds** [comment]
> It would probably be best to just leverage service provisioning/binding.

I don't know what that means.

> I'd like to get rid of templates that just hook 2 things together.

Ah, then we shouldn't include these.

> Plus a -1 on templates like https://raw.githubusercontent.com/redhat-developer/s2i-dotnetcore/master/templates/dotnet-runtime-example.json
as it uses Docker build strategy, this won't work online.

+1 on catching this. I'll do a PR to remove it.

**Comment by tmds** [comment]
> Here's some high level info and links to the service https://blog.openshift.com/whats-new-in-openshift-3-7-service-catalog-and-brokers/

@sspeiche I'm giving this a try. One thing I notice is I don't get an overview of 'environment variables' to change the build process. In contrast, when using the template, the UI gives a list of the template parameters, which correspond to s2i environment variables for the simple templates. So the UX for the simple template is nicer than for the imagestream.

> And then we'll also have to get it into openshift-ansible for it to ship w/ OCP.

@bparees They are already in OCP. I'll do a PR to add them to origin as well.

In general, when we make changes to templates/imagestreams in s2i-dotnetcore, should we make PRs to update openshift-ansible/origin too?

**Comment by bparees** [comment]
> In general, when we make changes to templates/imagestreams in s2i-dotnetcore, should we make PRs to update openshift-ansible/origin too?

yeah unfortunately that is the process currently.

**Pull request merged**